webapp/src/Controller/API/LanguageController.php

```diff
@@ -122,6 +123,84 @@ public function updateExecutableActions(Request $request, string $id): void
         $this->dj->auditlog('executable', $language->getLangid(), 'updated');
     }
 
+    #[IsGranted('ROLE_ADMIN')]
+    #[Rest\Post('languages')]
+    #[OA\Response(
+        response: 200,
+        description: 'Configure all specified languages.',
+    )]
+    public function configureLanguagesAction(Request $request): Response
+    {
+        /** @var UploadedFile|null $jsonFile */
+        $jsonFile = $request->files->get('json');
+        if (!$jsonFile) {
+            throw new BadRequestHttpException('No JSON file supplied.');
+        }
+        $newLanguages = $this->dj->jsonDecode(file_get_contents($jsonFile->getRealPath()));
+
+        // Disable submission for all current languages, we will enable it for all new languages below.
+        $curLanguages = $this->em->getRepository(Language::class)->findAll();
+        foreach ($curLanguages as $language) {
+            /** @var Language $language */
+            $language->setAllowSubmit(false);
+        }
+
+        $idField = $this->eventLogService->externalIdFieldForEntity(Language::class) ?? 'langid';
+        foreach ($newLanguages as $language) {
+            /** @var Language $language */
+            $lang_id = $language['id'];
+            $lang = $this->em->getRepository(Language::class)->findOneBy(
+                [$idField => $lang_id]
+            );
+            if (!$lang) {
+                continue;
+            }
+            $lang->setAllowSubmit(true);
+            if (isset($language['name'])) {
```

Comment:
We could use a PropertyAccessor loop here to get rid of most duplication. See also the problem import somewhere (I believe)

Reply:
I tried this and it was not really less code / more understandable (probably also because of the nested compiler/runner fields). Feel free to submit to this branch.

Reply:
Then I’m fine as is